integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -193,6 +193,168 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
           });
         }).to.throw("Only manual client resets are supported with flexible sync");
       });
+
+      describe("initialSubscriptions option", function () {
+        function getConfig(
+          user: Realm.User,
+          initialSubscriptions: Realm.FlexibleSyncConfiguration["initialSubscriptions"],
+        ): Realm.Configuration {
+          return {
+            schema: [FlexiblePersonSchema, DogSchema],
+            sync: {
+              _sessionStopPolicy: SessionStopPolicy.Immediately,
+              flexible: true,
+              user,
+              initialSubscriptions,
+            },
+          };
+        }
+
+        describe("error", function () {
+          afterEach(function () {
+            Realm.deleteFile(this.config);
+          });
+
+          it("throws an error if no `update` function is provided", async function () {
+            this.config = getConfig(this.user, {} as any);
```

Comment:
This suggestion applies to a couple of other places in this file.

```suggestion
            @ts-expect-error Intentionally testing the wrong type
            this.config = getConfig(this.user, {});
```

Reply:
👍